DotNetHowToSecurityTrimming/DotNetHowToSecurityTrimming/MicrosoftGraphHelper.cs

```diff
@@ -0,0 +1,205 @@
+        private string _clientId;
+        private string _token;
+
+        public MicrosoftGraphHelper(string clientId)
+        {
+            _clientId = clientId;
+        }
+
+        public void CreateGraphServiceClient()
+        {
+            PublicClientApplication app = new PublicClientApplication(_clientId);
+            string[] scopes = { "User.ReadWrite.All", "Group.ReadWrite.All", "Directory.ReadWrite.All" };
+
+            // Instantiate the Microsoft Graph, and provide a way to acquire the token. If token expires, it will
+            // be acquired again
+            _graph = new GraphServiceClient(new DelegateAuthenticationProvider(
+            async (requestMessage) =>
+            {
+                AuthenticationResult result = null;
+                // If a user has already signed-in, we try first to acquire the token silently, and then if this fails
+                // we try to acquire it with a user interaction.
+                var user = app.Users.FirstOrDefault();
+                if (user != null)
+                {
+                    try
+                    {
+                        // Attempts to acquire the access token from cache
+                        result = await app.AcquireTokenSilentAsync(scopes, user);
+                    }
+                    catch (MsalClientException ex)
+                    {
+                        if (ex.ErrorCode == "interaction_required")
+                        {
+                            // Interactive request to acquire token
+                            result = await app.AcquireTokenAsync(scopes);
+                        }
+                    }
+                }
+                else
+                {
+                    result = await app.AcquireTokenAsync(scopes);
+                }
+                _token = result.AccessToken;
+                requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", _token);
+            }));
+        }
+
+        private string BuildBatchRequest(List<string> users)
+        {
+            string requestBodyFormat = @"{{ ""requests"": [ {0} ]  }}";
+            string requestsBodyFormat = @"{{
+                    ""id"":""{0}"",
+                    ""method"":""POST"",
+                    ""url"":""users/{1}/microsoft.graph.getMemberGroups"",
+                    ""body"": {
+                        ""securityEnabledOnly"":true
+                    },
+                    ""headers"": {
+                        ""Content-Type"":""application/json""
+                    }
+                }},";
+
+            string requestsBody = null;
+            for (int i = 0; i < users.Count; i++)
+            {
+                requestsBody += string.Format(requestsBodyFormat, i, users[i]);
+            }
+            return string.Format(requestBodyFormat, requestsBody);
+        }
+
+        public async Task<string> SendRequestAndGetResponse(HttpClient client, List<string> users)
```

Comment:
I think this method should return the user-> groups dictionary (Dictionary<string, <string>>) so that the caller doesn't have to know how to parse the response